src/main/java/io/icker/factions/mixin/ServerPlayerEntityMixin.java

```diff
@@ -30,14 +30,14 @@ protected ServerPlayerEntityMixin(EntityType<? extends LivingEntity> entityType,
 
     @Inject(at = @At("HEAD"), method = "setClientSettings")
     public void setClientSettings(ClientSettingsC2SPacket packet, CallbackInfo info) {
-        User member = User.get(((ServerPlayerEntity)(Object) this).getUuid());
+        User member = User.get(getUuid());
         member.language = packet.language();
     }
 
     @Inject(at = @At("HEAD"), method = "onDeath")
     public void onDeath(DamageSource source, CallbackInfo info) {
         Entity entity = source.getSource();
-        if (entity == null || !entity.isPlayer()) return;
+        if (entity == null || !entity.isPlayer()) PlayerEvents.ON_PLAYER_DEATH.invoker().onPlayerDeath((ServerPlayerEntity) (Object) this, source);
```

Comment:
We might want to combine ON_PLAYER_DEATH and ON_KILLED_BY_PLAYER and pass the damamge source to the listeners and/or a boolean value of if it was a player.

Reply:
I agree, when typing this code I was thinking about making two different events just in case (kinda like a normal event api)...
Will do it in the following days (currently on vacation, I don't always have time to work on code)

Reply:
Since its a relatively small change I just did it myself